packages/core/src/UserState.ts

```diff
@@ -88,14 +87,20 @@ export default class UserState {
         const currentEpoch = await this.sync.loadCurrentEpoch()
         let latestTransitionedEpoch = 0
         for (let x = currentEpoch; x >= 0; x--) {
-            const epkNullifier = genUserStateTransitionNullifier(
-                this.id.secret,
-                this.sync.attesterId.toString(),
-                x
+            const nullifiers = [
```

Comment:
I removed the old epoch transition nullifier because we can use the epoch keys that are output by the UST proof. We can use just the first one, it will be either epoch key with nonce 0, or nonce `MAX_NONCE`, depending on if the epoch key received an attestation.

It should be safe to use as the transition nonce because for a certain epoch/epoch tree combination it will always be the same.